### PR TITLE
flux-plugin: Enable/Disable force reconciliation

### PR DESCRIPTION
**Pull request by ashu8912**

### CODE REVIEW

**Comment by ashu8912** [comment]
fixes #122

File: flux/src/actions/index.tsx [hunk 1]
```diff
@@ -3,6 +3,123 @@ import { KubeObject } from '@kinvolk/headlamp-plugin/lib/lib/k8s/cluster';
...
+            });
+        }}
+        title={'Enable Force Reconciliation'}
+        description={`Are you sure you want to enable force reconciliation?`}
+      />
+    </>
+  );
+}
+
+function DisableForceReconciliation(props) {
```
Comment (joaquimrocha) [inline flux/src/actions/index.tsx]: Seems like this component is the same as above, just with different logic + strings based on whether it's enable/disable. I think we should just have one single component.

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
Besides making just one component, the commit should be prefixed with the plugin name `flux:`, not `flux-plugin:`.

File: flux/src/actions/index.tsx [hunk 2]
```diff
@@ -3,6 +3,66 @@ import { KubeObject } from '@kinvolk/headlamp-plugin/lib/lib/k8s/cluster';
 import { useSnackbar } from 'notistack';
 import React from 'react';
 
+function ForceReconciliationAction(props) {
+  const { enqueueSnackbar } = useSnackbar();
+  const [open, setOpen] = React.useState<boolean>(false);
+  const { resource } = props;
+  if(!resource.jsonData.spec.force) {
+    return null;
+  }
```
Comment (joaquimrocha) [inline flux/src/actions/index.tsx]: This will prevent the component from rendering if the force is false.

File: flux/src/actions/index.tsx [hunk 3]
```diff
@@ -3,6 +3,66 @@ import { KubeObject } from '@kinvolk/headlamp-plugin/lib/lib/k8s/cluster';
...
+                force: false,
+              },
+            },
+            resource.jsonData.metadata.namespace,
+            resource.jsonData.metadata.name
+          )
+            .then(response => {
+              if (!response.spec.force) {
+                enqueueSnackbar(
+                  `Successfully ${forceReconciliationType}d force reconciliation for ${resource.metadata.name}`,
```
Comment (joaquimrocha) [inline flux/src/actions/index.tsx]: As a general rule, never generate words for human readable messages like this. Just use `Disabled force reconciliation for ${resource.metadata.name}` and `Enabled force reconciliation for ${resource.metadata.name}`

File: flux/src/actions/index.tsx [hunk 4]
```diff
@@ -3,6 +3,66 @@ import { KubeObject } from '@kinvolk/headlamp-plugin/lib/lib/k8s/cluster';
...
+                  { variant: 'success' }
+                );
+              } else {
+                enqueueSnackbar(`Failed to ${forceReconciliationType} force reconciliation for ${resource.metadata.name}`, {
```
Comment (joaquimrocha) [inline flux/src/actions/index.tsx]: Same about not generating english words from variables.

File: flux/src/actions/index.tsx [hunk 5]
```diff
@@ -3,6 +3,66 @@ import { KubeObject } from '@kinvolk/headlamp-plugin/lib/lib/k8s/cluster';
...
+                  variant: 'error',
+                });
+              }
+            })
+            .catch(error => {
+              enqueueSnackbar(
+                `Failed to ${forceReconciliationType} force reconciliation for ${resource.metadata.name} error ${error}`,
```
Comment (joaquimrocha) [inline flux/src/actions/index.tsx]: `Failed to enable force reconciliation for ${resource.metadata.name}: ${error}`
and
`Failed to disable force reconciliation for ${resource.metadata.name}: ${error}`

File: flux/src/actions/index.tsx [hunk 6]
```diff
@@ -3,6 +3,66 @@ import { KubeObject } from '@kinvolk/headlamp-plugin/lib/lib/k8s/cluster';
...
+                { variant: 'error' }
+              );
+            });
+        }}
+        title={`${forceReconciliationType} Force Reconciliation`}
+        description={`Are you sure you want to ${forceReconciliationType} force reconciliation?`}
```
Comment (joaquimrocha) [inline flux/src/actions/index.tsx]: ditto

**Review by joaquimrocha — CHANGES_REQUESTED** [review]

**Pull request merged**